
%%%% MACRO DEFINITION %%%%

\providecommand{\pvivax}{P.~vivax}
\providecommand{\pfalciparum}{P.~falciparum}
\providecommand{\cterm}{C-terminus}
\providecommand{\nterm}{N-terminus}

\providecommand{\e}[1]{\ensuremath{\times 10^{#1}}}
\newcolumntype{P}[1]{>{\centering\arraybackslash}p{#1}}
\newcolumntype{M}[1]{>{\centering\arraybackslash}m{#1}}

\providecommand{\refimage}[1]{\figurename~\ref{fig:#1}}

% figure that is as wide as the text
\newcommand{\insertfigure}[5][1]{
\begin{figure}[h!]
	\makebox[\textwidth]{\includegraphics[width=#1\textwidth]{Chapter1_pics/#2}}
	\caption[#3]{\small{#4}}\label{fig:#5}
\end{figure}
}

%TC:macro \note [ignore]



%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%													BEGIN
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\chapter{Introduction} % Write in your own chapter title
\label{Chapter1}
\lhead{Chapter 1. \emph{Introduction}} % Write in your own chapter title to set the page header

Proteins are polymer chains of amino acid residues, and while the chemical diversity of the twenty canonical amino acids utilised in Biology can offer proteins a staggering variety of folds and functionality, there remain chemical processes that are not possible using only this chemical species. Consequently, many proteins use cofactors. These are chemicals which associate with the protein but are not generally covalently bound to them.

In order for a protein to induce this association to happen, it must present a region of its surface that the cofactor in question will experience an attraction to, and for which association will be thermodynamically favourable. The residues that make up this attractive region are called binding sites.

Metal atoms are a common cofactor. In this case the binding site is a metal binding site, and when the metal is zinc we can speak of a `zinc binding site'.

Clearly, zinc will only experience an attraction towards certain kinds of protein surfaces, and so proteins which need to attract a zinc cofactor will have to present surface residues capable of doing this. It should be possible to predict, therefore, whether a protein's structure has a potential zinc binding site on it. Furthermore, because structure is determined by amino acid sequence, it should in principle be possible to predict zinc binding capabilities from protein sequence.

This PhD project is an attempt to develop novel methods for doing so.

\section{Zinc}

\subsection{Cofactors}

Proteins are an expression of the will of genetic material, and are the means by which the information content of that material is effected. The translation process maps three-nucleotide codons to amino acids, building up proteins as polypeptide chains of some combination of twenty of these building blocks. These amino acids all contain the same backbone, but each has its own unique side chain, and between them the twenty amino acid side chains cover a diverse range of chemical properties. There are hydrophobic and hydrophilic side chains, side chains of varied length, and electronic charge.

However, while this `protein-space' is certainly vast, the different possible combinations and orientations of the twenty amino acid side chains still represent a small subspace of total chemical space, and there are limits to what proteins can achieve using only these chemical species. The principal atoms involved - carbon, nitrogen, oxygen and sulphur - are all clustered at upper right corner of the periodic table, all have broadly similar electronic properties, and are all similar in size.

Other sections of the periodic table are not available to evolution by the direct means of encoding them in DNA - there is no codon that will cause them to be incorporated directly - but these other chemical species can be incorporated through other means. The three-dimensional structure of a protein is determined by primary structure, so the genetic code can create a protein which presents a region of its surface that the `desired' (in the evolutionary sense) chemical species will be attracted to and will associate with the protein through non-covalent means - assuming it is present in the protein's environment. These are `cofactors'.

While there are many organic, molecular cofactors in use, metal ions - particularly transition metal ions - have particular properties that make them very useful as cofactors. To understand this, it will be necessary to briefly review the electronic structure of atoms, show how transition metals are distinguished by their particular electronics structure, and then look at how zinc's properties are unusual by the standards of other transition metals.

Atom's electrons are arranged into energy levels, which are in turn sub-divided into orbitals - regions of space in which electrons will usually be found. For example, atoms can hold two electrons in the first, lowest energy level, which has a single s-orbital. The second energy level can hold eight electrons, with one slightly lower energy s-orbital and three slightly higher energy p-orbitals. This describes the electron arrangement of the first ten elements.

The third energy level is slightly more complicated. It has three sub-levels - an s-orbital, three p-orbitals, and five d-orbitals, but the d-orbitals are actually in a higher energy state than the s-orbital of the \emph{fourth} energy level, which `fills up' first. The order in which the orbitals fill up is therefore 1s2s2p3s3p4s3d4p and so on. An atom with thirteen electrons (i.e. aluminium) will have its thirteenth electron in a 3p orbital, an atom with nineteen electrons (potassium) will have its nineteenth electron in a 4s orbital, and only once an atom has to `add' a twenty-first electron (scandium) will the 3d orbitals be used.

Atoms which use these d orbitals have unusual properties not encountered in previous atoms as a consequence.

\subsection{Transition Metals}

Transition metals are often employed as cofactors by proteins, particularly in catalysis. They occupy a different region of electron configuration space than the organic, non-metal atoms that make up the biological amino acids, and so can provide functionality that would otherwise be unavailable.

Atom's `prefer' to have full energy levels and sub-energy levels - in the sense that if they are in an environment which draws electrons away from it (an oxidising environment) it will be energetically favourable to lose whatever number of electrons causes it to fall back to the last filled level, emptying the highest energy level. Likewise, if in an environment in which electrons are pushed to the atom (a reducing environment), they will prefer to take the number of electrons which fills the current energy level.

For elements below scandium, this is straightforward - energy levels are far apart, and there is generally only one clear point to fall back to. Sodium for example, has a single 3s electron, so when in an oxidising environment it can relinquish this one electron, and become Na+. It has a single oxidation state.

Once d-orbitals start being filled however, this becomes less straightforward. Rather unintuitively, when it comes to /emph{losing} electrons, the 4s electrons are the highest energy, and these will be lost first. Therefore most atoms with unfilled d-orbitals can adopt the 2+ oxidation state. The d-orbitals are rather close in energy level, so if there is further oxidative pressure, these can be lost too. This property of having multiple stable oxidation states is one of the chief characteristics of transition metals. It is partly for this reason that transition metals are so widely employed by evolution as protein cofactors, particularly in proteins which catalyse reactions - enzymes. A common mechanism for catalysing a reaction is to stabilise an otherwise very thermodynamically unfavourable intermediate that might have a strong charge associated - by using a chemical species which can have one oxidation state in its native state but which will happily switch to a more oxidised form, the excess charge from the reaction intermediate can be temporarily `stored' in the cofactor, stabilising the intermediate.

As cations, the transition metals also act as efficient Lewis acids - they can accept lone pairs of electrons. This is also useful in catalysis as it can stabilise intermediate structures by withdrawing excessive negative charge from them. This achieves a similar effect, but do so using a different mechanism. The metal atom is not being oxidised, 

Another curious property of atoms with partially filled d-orbitals is the so-called breaking of degeneracy. In a lone atom of such elements, the five d-orbitals are said to be degenerate, in the sense of having identical energy levels. This is also true when oxidised, but when these charged ions interact with a solvent molecule or other ligands, some d-orbitals end up closer than others to the orbitals of the incoming ligand, entering a higher energy state than the others. The previously uniform d-orbitals split into two slightly different energy levels, and the previous degeneracy is said to have broken.

This confers some unique properties on these elements - most visibly that of colour. The electrons will prefer to occupy the lower energy d-orbitals, but the energy gap between them is small enough as to be in the visible light range, so light passing through a solution of these ions will absorb some wavelength of them.

It also means that different ligand geometries are associated with different changes in energy, meaning that certain geometries end up being preferred over others. This has implications for proteins, which must fold in such a way that the liganding residues adopt the correct geometry.





\note{Also explain how ligand/crystal stabilisation energy works.}

Unsurprisingly then, many proteins have evolved to take advantage of these useful properties of transition metals by presenting binding sites on their surface that will acquire one of them. This is generally done by bringing residues with available lone pairs into close proximity to each other, in an arrangement that matches the dimensions and orbital geometry of the desired metal.

\subsection{Zinc's Unique Properties}

Though a transition metal in the sense that it ten d-electrons but no 4p electrons, it is an unusual one - to the point where it is often not even classified as one. Many of the properties of transition metals derive from \emph{partially filled} d-orbitals, which zinc does not have. The five fully filled d-orbitals are relatively stable, meaning it can only really be oxidised to 2+ by losing its 4s electrons, so it has just one oxidation state. Electrons cannot be `promoted' from one d-orbital to another in solution as all spaces in the orbitals are already filled, so it has no colour or spectroscopic activity when in solution. At first glance, zinc would appear to be rather unremarkable metal, unworthy of much consideration by evolution.

However, the data show otherwise. About 10\% of all proteins in the human genome are zinc proteins with a zinc binding site - the second most abundant such metal after iron. It is also the only metal found in all classes of enzyme. Clearly, evolution has found zinc very useful.

In fact it is precisely that unremarkableness that has made zinc so attractive. Having just one oxidation state may mean that it cannot hold onto an electron mid-reaction, but it also means that the ion will retain its electronic properties in a wide range of reducing environments - and its status as a particularly good Lewis acid means it is still very useful in catalysis. Its lack of spectroscopic activity may make zinc solutions colourless, but the same arrangement of d-orbitals that causes this also means that there is no energetic penalty for zinc coming out of solution (where it is octahedrally coordinated) and into the binding site \note{Explain this MUCH better}.

\section{Properties of Zinc Binding Sites}

What do zinc binding sites `look like' from a structural point of view? What are their properties at the atomic scale?

These questions were addressed by researchers from the very first crystal structures, and continue to be a topic of research today. The properties of zinc binding sites that distinguish them from other spatially proximate clusterings of residues are ultimately what must be used to predict them, so the advances in this field are crucial to any zinc binding prediction model. The history of our understanding of these properties will be reviewed here.

By the middle of the 1980s, a number of structures had been produced and already a few general themes that would recur over and over again were observed. Catalytic sites tended to have three protein ligands and one water ligand, whereas structural sites generally had four protein residues, for example. One of the first reviews in this area to examine the structures obtained so far identified coordination by sulphur and nitrogen, and tetrahedral geometry, as the two defining characteristics of zinc binding sites \cite{williams1987biochemistry}. With the benefit of hindsight we can see that this is a slight simplification, but it shows that a consensus about `typical features' was beginning to form. Another early review would go on to list most of the basic properties of zinc binding sites we now know - their division into structural and catalytic sites, the near ubiquity of water in catalytic sites, the much stronger preference for tetrahedral geometry in structural sites than in catalytic sites, and the preference for histidine, cysteine, aspartate and glutamate residues \cite{vallee1990zinc}.

These properties would be repeated and elaborated upon by a number of similar reviews in this period \cite{tainer1991metal,vallee1992functional,coleman1992zinc}.

The more structures that were available, the more detailed the inferences that could be made became. A series of reviews looked at typical atom geometries in cysteine binding \cite{chakrabarti1989geometry} and histidine binding \cite{chakrabarti1990geometry}, and while both papers looked at metal binding generally rather than zinc, in both cases the observations were found to be largely metal-agnostic. It was shown that cysteine generally coordinated the metal such that the Zn---S-C-C torsional angle is either 90$^\circ$ or 180$^\circ$ , and that histidine residues generally coordinate the metal via their NE2 nitrogen atom, with the metal lying in the histidyl plane.

Another structural characteristic of zinc binding sites - and indeed metal binding sites generally - was the `hydrophobic contrast' that seemed to exist around them \cite{yamashita1990metal,gregory1993prediction}. Briefly, this is the observation that metal atoms in a protein tend to be surrounded by a shell of hydrophilic atoms (as might be expected), but that these were in turn surrounded by hydrophobic atoms. This could even be implemented as a function of coordinates, which would be at a maximum when centered in such a concentric sphere. It was shown that maxima of this function would cluster around metal binding sites.

Whilst it might be expected that zinc binding sites would have some kinds of structural consensus, it is not immediately obvious that they should have any kind of predictable sequence patterns. However in 1989 it was shown, albeit from a relatively small dataset as existed at that time, that catalytic zinc binding sites seemed to have characteristic ``short and long spacer sequences" \cite{vallee1989short}. That is, the three residues that make up catalytic binding sites seem to be made up of two residues separated by a small stretch of amino acids - around five - and a third residue that is much more distal on the sequence. They proposed that the first two residues were acting as a nucleus, a proto-site that is stabilised by the third residue. This pattern was confirmed by later studies, with the short spacer being found to be generally 2 to 7 residues long \cite{patel2007analysis}.

This was not however the \textit{first} sequence motif associated with zinc binding sites - a few years previously a curious pattern of residues in a transcription factor sequence led to the discovery of a new class of zinc protein, and the creation of a entirely new field of genome engineering.

\note{Explain zinc fingers}

The field continued to grow - by 1998 there was thought to be at least 500 zinc finger proteins, and possibly as many as 1\% of all mammalian genes \cite{mackay1998zinc}. Much of this field is concerned with the `fingertip' end of zinc fingers, which actually implements the DNA-recognising properties, though for the purposes of this project it is the base that is of more interest. There are a few points to takeaway from this brief digression. The first point, which will be returned to in the section of zinc binding site prediction, is that the zinc fingers were discovered because of a characteristic \textit{sequence} pattern, emphasising that such prediction is both possible and potentially offers many insights. The second is that until this point, much of the interest in zinc binding sites was in catalytic zinc binding sites - there were fewer structural sites available to study, and most reviews from this period addressed them last, if at all. From this point on however, many more became available to study, and it became clear that they were responsible for much of zinc's ubiquity in the proteome.

Most of the early work on characterising the typical properties of zinc binding sites was done by manually surveying the crystal structures available and drawing conclusions from them. But as the 1990s wore on and the number of such structures available reached the hundreds, this became increasingly impractical. Instead, from this point onwards, the sites would have to be collated algorithmically from the Protein Data Bank.

The first such database survey was in 1998 - they found 387 zinc proteins, and after excluding some on quality and experimental technique factors, they were left with 111 \cite{alberts1998analysis}. Some of their findings were in line with what was already known - the three residue/one water combination for catalytic sites, the preference for tetrahedral geometry, histidine's preference for one nitrogen over another (by a ratio of 70:30, they found). They also identified a new almost ubiquitous motif across zinc binding sites which they call the `elec-His-zinc' motif whereby zinc is liganded by a histidine which in turn forms a hydrogen bond with an electron donor of some kind.

This was followed by similar reviews \cite{roe1999zinc,laity2001zinc,grishin2001treble,harding2001geometry,krishna2003structural}, showing similar results - that zinc-ligand distances are similar to those found in small molecules, and further sub-categorising the ever increasing number of zinc finger proteins.

Increasingly, the modes of classification were becoming more sophisticated too. Initially, two kinds of binding site were recognised - catalytic sites, which catalysed some metabolic reaction, and structural sites, which did not and just stabilised structure. Later came the recognition that some sites required the action of multiple metals in a single functional unit - either multiple zinc atoms, or a zinc atom in conjunction with another metal atom. From 2001, a fourth kind was recognised - interface binding sites \cite{auld2001zinc}. It had been known since the early days of insulin research of course that in some cases zinc atoms could stabilise polymeric structures, but by this point there was enough data to classify these as their own category, with their own properties - that cysteine is very underrepresented in them, that stabilisation is often done by beta sheets rather than alpha helices (common in other zinc binding sites), even evidence that these sites could also have catalytic roles in addition to their role in quaternary structure.

A more in-depth look at the differences between catalytic and structural sites even seemed to offer a means of automatically classifying zinc binding sites \cite{lee2008physical}, though generally during this period this was a task done manually. They confirmed that catalytic zinc binding sites tended to have more unusual atom distances and geometries (imposing a `strain' upon them that we will return to in the following section), and looked in detail at catalytic sites' disdain for cysteine residues. They found that a relatively simple method of assuming a site is structural if it is has more than one cysteine residue \textit{and} no water, and assuming it is catalytic otherwise, assigned the correct label in 90\% of cases in their dataset. They also looked at the possibility of using atom distances to classify zinc binding sites, though it must be remembered that when looking at differences in atom distance on the order of single Angstroms, the resolution of the structure you are examining is critical if you are to make sensible inferences.

Atom distances themselves have been a major focus of research. A typical example is a 2007 review \cite{tamames2007analysis} of 994 structures, showing that histidine approaches zinc more closely than cysteine (2.11 \AA \ versus 2.32 \AA \ on average) - with water distances generally somewhere between those two figures. The vast majority of metal liganding atoms are within 3 Angstroms of the metal \cite{dokmanic2008metals}.

The properties of zinc binding sites are not just limited to the metal itself and the residues that coordinate it. It is increasingly recognised that the `secondary shell' around the primary residues are of great importance, by providing a network of stabilising hydrogen bonds around the liganding residues to keep them in their precise orientation \cite{auld2001zinc}. This concept has been extended to the idea of a `Minimal Functional Site' \cite{andreini2011minimal}, proposed as the smallest unit of metal binding that should be used when classifying and studying the phenomenon. The authors in this case used all residues within 5 \AA \ of a liganding residue, but more important is the general idea that the residues around the coordinating residues, which are often part of secondary structure - are crucial to understanding the metal binding itself.

There have also been reviews into the efficacy of these database reviews at all, and the limits of what can be inferred from them. A 2005 review \cite{sommerhalter2005x} suggests that while purely spatial properties, such as the location of the metal ion and its liganding residue identities, can be provided reliably by crystallography, often the precise coordination geometry does not agree with spectroscopic data, and should be treated with caution. They also warn against trying to infer redox state from crystallography, though this does not particularly apply to zinc. A later review \cite{laitaoja2013zinc} pointed out that reported atom distances increase as a function of resolution, and stresses the need to take this into account. They criticise a number of earlier studies for treating unlikely geometries or electronic configuration as `real', or for treating zinc salts and crystallographic artefacts as biologically relevant zinc binding sites.

Finally, it is worth examining the numerous secondary databases that have been created over the past decade and a half relating to zinc binding sites, and which offer a web-accessible means of viewing the properties of zinc (though usually metals generally) binding sites.

MESPEUS was one of the first - a general-purpose database of metal binding sites generally, which focused on providing basic geometric information \cite{hsin2008mespeus}, and which allowed basic surveys of atom-length patterns to be carried out.

ZifBASE had a more specific focus - this was a database of zinc fingers specifically, and provided much more than structural information from the PDB - sequence information, DNA targets, and physiochemical properties like isoelectric point were also provided \cite{jayakanthan2009zifbase}.

Probably the most important resource in this area is MetalPDB \cite{andreini2012metalpdb}, a database of metal binding sites, similar to MESPEUS but with more information and a more sophisticated means of classifying them. They divide their sites among `equistructural sites' - those which have binding sites of similar structure, and within those `equivalent sites' - equistructural sites with the same metal. They place a great focus on the secondary shell, as part of the `Minimal Functional Site' concept they introduced earlier, and have recently updated the resource with a greater focus on apo-structures \cite{putignano2017metalpdb}. They have also provided a number of related resources, such as MetalS3 \cite{valasatava2014metals}, a tool for searching the database via structure pattern rather than keyword, and tools for classifying sites based on similarity \cite{sobolev2013web}.

An issue with MetalPDB however is its reliance on asymmetric units in the PDB files they parse - often the raw coordinates of such files represent only the repeating unit of the crystal structure, and the actual `real' biological protein must be assembled from that data using transformation matrices contained with the files. MetalPDB (and most other resources) don't do this, which makes some of their `binding sites' rather unrealistic.

Another impression one gets immediately when surveying these resources is that with the exception of MetalPDB, all of these resources have ceased existing - their URLs go nowhere, or their sites are broken beyond reasonable use and obviously not maintained. This is a problem generally with academic software.

Other resources are focused more on validating or describing a given metal binding site that it is presented with, such as the coordination geometry classifier FindGeo \cite{andreini2012findgeo}, or the validation tool CheckMyMetal \cite{zheng2017checkmymetal}, which uses known properties of metal binding sites to determine if a proposed binding site is realistic.


\section{Predicting Zinc Binding}

There are still many proteins whose function is unknown. In many cases we only know their sequence, though some of these have had their structure solved experimentally.

One means of determining what function a protein might be intended to perform is to look for certain motifs associated with a given function. If, for example, it can be shown that such a protein had one or more zinc binding sites, that could offer insights into what function the protein performs. Indeed, that is ultimately the goal of this project.

Owing to this clear usefulness, there have been a number of attempts to develop such predictors in the past, dating back to the early 1990s. These methods and techniques form the essential context in which my own project should be understood, and also provide the benchmark of success against which my methods should be judged. Here their progress will be reviewed. They can essentially be divided into two broad categories - prediction from structure and prediction from sequence.

\subsection{Prediction from Structure}

Predicting zinc binding from structure takes the atom coordinates of some protein as its input, and produces as output residues in that structure which are believed to be able to bind zinc if zinc were present - generally along with some estimated probability that this should be the case.

This is perhaps of only limited utility, as very often proteins which can bind zinc tightly will have zinc bound when the structure is solved, and so no prediction is required. Where such methods are of use however is when a protein capable of binding zinc is crystallised without zinc being present - presumably because the researchers didn't know it could bind zinc and so did not provide any in the medium. These apoproteins can be scanned by predictors like this to identify unfilled zinc binding sites. \note{NMR invisibility?}

\note{Other uses?}

The earliest attempt at a general purpose prediction algorithm was the Hydrophobic Contrast Function, developed in 1990 \cite{yamashita1990metal} to detect metal binding sites in general. This began from the observation that metal atoms tended to be surrounded by a shell of hydrophilic atoms, which in turn were surrounded by an outer shell of hydrophobic atoms. They developed a function which took as its input a coordinate in space, and returned a measure of this `hydrophobic contrast' at that location. They showed that by scanning every point in the structure on a grid, evaluating the function at every point, and then ranking the points by their score (the function returned values between -1 for inverted contrast and 1 for desired contrast), the highest scoring points would be clustered at sites of metal binding.

This was the first demonstration that a property of metal binding sites could be used to create a predictive model. However it should be noted that it was only tested on structures with metals already present, not on apoproteins, and the initial observation itself was based on a very small number of metal proteins that were available at that time. It was later expanded upon by combining it with a template-based searching algorithm, whereby the known metal binding sites were represented as stripped-down `templates' of just alpha and beta carbons, and used to search proteins using a simple pattern matching algorithm \cite{gregory1993prediction}. Once a set of three residues matching a known template was found, the hydrophobic contrast function was applied to verify that the binding site was a region of high contrast, which it generally was. This was part of a pipeline for engineering binding sites into proteins, so it didn't require the matching residues to be of any particular type, but the general principle of using known site geometry and hydrophobic contrast properties to look in new structures was still in use.

\emph{MetSite}

\emph{Fold-X}

\emph{Goyal and Mande}

\emph{CHED}

\emph{FEATRUE}

\emph{SitePredict}

One of the more recent algorithms for identifying potential zinc binding sites in protein structures is TEMSP \cite{zhao2011structure}. This is actually reminiscent of some of the earlier work on this problem, in that it uses simple geometric patterns of known zinc binding sites - specifically their alpha and beta carbons - to search target structures without any particular machine learning method employed (though there is a certain amount of parameter training). Here the problem is simplified by creating a library of `residue pairs', where for each zinc binding site obtained from the PDB, they store each pairwise combination of liganding residues (actually just certain properties of them, such as alpha-alpha distance etc.) and then search the target protein for these. Once one is found, other template pairs are superimposed on the structure to find additional liganding residues. They used a stricter definition of `true positive' than previous attempts, and still reported a sensitivity of 86.0\% and a selectivity of 95.9\% in their test dataset - barely lower than the values obtained when testing on the training dataset used to refine their parameters. When applying their tool to a dataset of proteins of unknown function, they found some very promising results - finding 186 probable zinc binding sites. \note{So what?} Unfortunately, the online version of this tool is no longer accessible at time of writing.

\subsection{Prediction from Sequence}

%\insertfigure[0.5]{name.eps}{short caption}{long caption}{label}
% N.B. do not \end{document} at the end of the chapters
